### PR TITLE
feat: ubuntu upgrade

### LINKED ISSUE #25: Ubuntu 24.04 upgrade
- Ensure CI uses the latest ubuntu
- Ensure dockerfile is updated to use ubuntu 24.04
- Ensure any documentation/references are updated

See https://github.com/overhangio/tutor/issues/1113 for context

### PR DESCRIPTION
**Pull request by Abdul-Muqadim-Arbisoft**
close #25

### CODE REVIEW

File: .github/workflows/auto-add-to-project.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     name: Add issue and bugs to project
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/add-to-project@v0.5.0
```
Comment (regisb) [inline .github/workflows/auto-add-to-project.yml:14]: Let me update this across all plugins right away. We'll upgrade straight to 1.0.2: https://github.com/actions/add-to-project/releases/tag/v1.0.2
Same for actions/checkout@v4 below.

**Review by regisb — COMMENTED** [review]
Please take this PR out of draft mode once it's ready. In the meantime I'll update the version of CI actions across all plugins.

**Review by DawoudSheraz — COMMENTED** [review]
please add a changelog-entry.

File: changelog.d/20241025_191448_abdul.muqadim_ubuntu_upgrade.md [hunk 2]
```diff
@@ -0,0 +1 @@
+-💥[Feature] Update android Image to use Ubuntu `24.04` as base OS. (by @Abdul-Muqadim-Arbisoft)
```
Comment (DawoudSheraz) [inline changelog.d/20241025_191448_abdul.muqadim_ubuntu_upgrade.md]: nit: android image

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**